.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, dev ]
+  pull_request:
```

Comment:
> Since CodeQL can take time to execute, it's not advisable to run during PR.

Reply:
So do we need both?

Reply:
I see.
The template changed from my last time.
The previous was:
```
on:
  push:
    branches: [ main ]
  schedule:
    - cron: '30 23 * * 0'
```

So, in the case that I was imagining, it wasn't running during PRs.
Based on that we can:
either remove LGTM or remove pull_request analysis from CodeQL.

Reply:
I'll defer to your recommendation.